Sources/Workspace/PinsStore.swift

```diff
@@ -182,19 +182,34 @@ extension PinsStore: SimplePersistanceProtocol {
 
 // JSON.
 extension PinsStore.Pin: JSONMappable, JSONSerializable, Equatable {
+
     /// Create an instance from JSON data.
     public init(json: JSON) throws {
+
+        // If it's a relative path then convert it into
+        // absolute path relative to the woking directory.
+        let repoUrl: String = try json.get("repository")
+        let absPath = AbsolutePath(repoUrl, relativeTo: currentWorkingDirectory).asString
+
         self.package = try json.get("package")
-        self.repository = try json.get("repositoryURL")
+        self.repository =  RepositorySpecifier(url: absPath)
         self.reason = json.get("reason")
         self.state = try json.get("state")
     }
 
     /// Convert the pin to JSON.
     public func toJSON() -> JSON {
+
+        // If it's a local path then convert it into
+        // relative path relative to the working directory.
+        var repositoryPath = repository.url
+        if URL.scheme(repositoryPath) == nil {
+            repositoryPath =  AbsolutePath(repositoryPath).relative(to: currentWorkingDirectory).asString
```

Comment:
`currentWorkingDirectory` can be made a parameter to init of PinStore with a default value, so it can be tested better.